### PR TITLE
DATAUP-497: more search and replace

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Final batch of job name standardisations.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-497
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [ ] Not ready for narrative testing yet!

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

### CODE REVIEW

File: docs/design/job_architecture.md [hunk 1]
```diff
@@ -366,7 +366,7 @@ Dictionary with key(s) job ID and value dictionaries with the following structur
 Logs data is split out into individual jobs and sent to `job_id` (see above)
 
 
-### `job_retries`
+### `retry_job`
```
Comment (ialarmedalien) [inline docs/design/job_architecture.md:369]: rename this for uniformity

File: docs/design/job_architecture.md [hunk 2]
```diff
@@ -265,7 +265,7 @@ These are described below. The name (`msg_type`) is given, followed by the keys
 
 By design, these should only be seen by the `JobCommChannel` instance, then sent into bus messages that get sent on specific channels. That information is also given in each block.
 
-### `job_comm_error`
+### `job_error`
```
Comment (ialarmedalien) [inline docs/design/job_architecture.md:268]: rename for consistency

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 3]
```diff
@@ -67,7 +67,7 @@ define([
         // commands
         CANCEL = 'cancel_job',
         CELL_JOB_STATUS = 'cell_job_status',
-        COMM_ERROR = 'job_comm_error',
+        ERROR = 'job_error',
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:70]: job_comm_error => job_error

File: src/biokbase/narrative/jobs/jobcomm.py [hunk 4]
```diff
@@ -23,6 +23,26 @@
 JOB_ID_LIST = "job_id_list"
 CELL_ID_LIST = "cell_id_list"
 
+# messages
```
Comment (ialarmedalien) [inline src/biokbase/narrative/jobs/jobcomm.py:26]: some nice constants in place of the message type names

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 5]
```diff
@@ -135,7 +135,7 @@ define(['common/errorDisplay', 'common/format', 'common/html', 'common/ui', 'uti
         const required = ['job_id', 'status'];
         return !!(
             jobState !== null &&
-            typeof jobState === 'object' &&
+            Utils.objectToString(jobState) === 'Object' &&
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:138]: as suggested in the previous PR, but mistakenly committed the edit in this branch

File: test/data/jobsData.js [hunk 6]
```diff
@@ -464,6 +464,13 @@ define(['common/format', 'testUtil'], (format, TestUtil) => {
             app_id: 'batch',
             job_params: [],
         },
+        {
+            job_id: 'some-crappy-job',
+            batch_id: null,
+            app_name: 'Some Crappy App',
+            app_id: 'some/crappy-app',
+            job_params: [{}],
```
Comment (ialarmedalien) [inline test/data/jobsData.js:472]: allow empty params

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 7]
```diff
@@ -198,14 +198,13 @@ define(['common/errorDisplay', 'common/format', 'common/html', 'common/ui', 'uti
...
-                typeof jobInfo === 'object' &&
+                Utils.objectToString(jobInfo) === 'Object' &&
                 required.every((key) => key in jobInfo) &&
                 Utils.objectToString(jobInfo.job_params) === 'Array' &&
                 // job_params is empty
                 (jobInfo.job_params.length === 0 ||
                     // job params is populated with an object
-                    (Utils.objectToString(jobInfo.job_params[0]) === 'Object' &&
-                        Object.keys(jobInfo.job_params[0]).length > 0))
+                    Utils.objectToString(jobInfo.job_params[0]) === 'Object')
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:207]: remove the requirement for more than zero job_params

File: src/biokbase/narrative/jobs/jobcomm.py (hunk 4 above)
Comment (ialarmedalien) [reply src/biokbase/narrative/jobs/jobcomm.py:26]: all changes in this file are either search-replace or minor documentation changes

File: src/biokbase/narrative/tests/test_jobcomm.py [hunk 8]
```diff
@@ -24,6 +24,21 @@
     JOB_ID,
     JOB_ID_LIST,
     CELL_ID_LIST,
+    CANCEL,
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/test_jobcomm.py:27]: all changes in this file are from the big message name search and replace

**Review by briehl — APPROVED** [review]
Find and replace PRs are the easiest to review!

**Pull request merged**